peer/network.go

```diff
@@ -175,6 +175,7 @@ func (n *network) SendAppRequest(nodeID ids.NodeID, request []byte, responseHand
 // Assumes write lock is held
 func (n *network) sendAppRequest(nodeID ids.NodeID, request []byte, responseHandler message.ResponseHandler) error {
 	if n.closed.Get() {
+		n.activeAppRequests.Release(1)
```

Comment:
I think we also need to release these during `Shutdown()`

Reply:
Although I don't think we have a way of knowning if we should be releasing from `activeAppRequests` or `activeCrossChainRequests` there 🤔

Reply:
Yeah you're right we have to release on `Shutdown` as well... I think we can actually do this by just making a wrapper handler type. I would not be surprised if this was the reason we had all those weird time out EVM bugs on shutdown...